apt/risk/data_assessment/dataset_attack.py

```diff
@@ -0,0 +1,111 @@
+"""
+This module defines the interface for privacy risk assessment of synthetic datasets.
+"""
+import abc
+from typing import Optional
+
+import matplotlib.pyplot as plt
+import numpy as np
+from sklearn import metrics
+from sklearn.metrics import RocCurveDisplay
+
+from apt.risk.data_assessment.attack_strategy_utils import AttackStrategyUtils
+from apt.risk.data_assessment.dataset_attack_result import DatasetAttackScore, DatasetAttackResultPerRecord
+from apt.utils.datasets import ArrayDataset
+
+
+class Config(abc.ABC):
+    """
+        The base class for dataset attack configurations
+    """
+    pass
+
+
+class DatasetAttack(abc.ABC):
+    """
+         The interface for performing privacy attack for risk assessment for synthetic datasets to be used in AI models.
+         The original data members (training data) and non-members (the holdout data) should be available.
+         For reliability, all the datasets should be preprocessed and normalized.
+    """
+
+    def __init__(self, original_data_members: ArrayDataset, original_data_non_members: ArrayDataset,
+                 synthetic_data: ArrayDataset, dataset_name: str, attack_strategy_utils: AttackStrategyUtils,
```

Comment:
I saw you changed dataset_name to optional in the assessment manager, there are a few more places to make this change.